### PR TITLE
Update to v3 step 1

### PR DESCRIPTION
**Pull request by slonka**

### CODE REVIEW

File: envoy-control-core/src/test/java/pl/allegro/tech/servicemesh/envoycontrol/v2/SimpleCacheTest.java [hunk 1]
```diff
@@ -107,17 +105,17 @@ public void invalidNamesListShouldReturnWatcherWithResponseInXdsMode() {
...
                         .addResourceNames("none")
-                        .build(),
+                        .build()),
                 Collections.emptySet(),
                 responseTracker,
                 false);
 
         assertThat(watch.isCancelled()).isFalse();
-        assertThat(responseTracker.responses).isNotEmpty();
+        Assertions.assertThat(responseTracker.responses).isNotEmpty();
```
Comment (pzmi) [inline envoy-control-core/src/test/java/pl/allegro/tech/servicemesh/envoycontrol/v2/SimpleCacheTest.java:118]: Looks like a mistake

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdaterTest.kt [hunk 2]
```diff
@@ -189,16 +191,18 @@ class SnapshotUpdaterTest {
...
-        assertThat(snapshot.routes().version()).isNotEqualTo(RoutesVersion.EMPTY_VERSION.value)
-
-        assertThat(snapshot.routes().resources().values).hasSize(2)
-        // two fallbacks: proxying direct IP requests and 503 for missing services
-        assertThat(snapshot.routes().resources().values
-            .first { it.name == "default_routes" }.virtualHostsCount)
-            .isEqualTo(2)
+        val clusters = snapshot.resources(Resources.ResourceType.CLUSTER)
+        assertThat(clusters).isNotNull()
+//        assertThat(clusters.version()).isEqualTo(ClustersVersion.EMPTY_VERSION.value)
```
Comment (pzmi) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdaterTest.kt:196]: Why is this commented?

**Review by pzmi — DISMISSED** [review]
LGTM. Just 2 nits

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdaterTest.kt (hunk 2 above)
Comment (slonka) [reply envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdaterTest.kt:196]: oopsie :( I'll revert that.

**Review by pzmi — DISMISSED** [review]

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/GroupChangeWatcher.kt [hunk 3]
```diff
@@ -41,12 +41,13 @@ internal class GroupChangeWatcher(
 
     override fun createWatch(
         ads: Boolean,
-        request: DiscoveryRequest,
-        knownResourceNames: MutableSet<String>,
-        responseConsumer: Consumer<Response>,
+        request: XdsRequest?,
+        knownResourceNames: MutableSet<String>?,
```
Comment (lukidzi) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/GroupChangeWatcher.kt:45]: Are `?` necessary?

Comment (slonka) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/GroupChangeWatcher.kt:45]: Good catch, should probably be reverted, probably intellij done this without my knowledge

**Review by lukidzi — APPROVED** [review]

**Review by pzmi — DISMISSED** [review]

**Review by lukidzi — DISMISSED** [review]

**Review by pzmi — APPROVED** [review]

**Review by lukidzi — APPROVED** [review]

**Review by pzmi — APPROVED** [review]

**Pull request merged**